src/sidebar/util/test/fetch-config-test.js

```diff
@@ -237,6 +239,52 @@ describe('sidebar.util.fetch-config', () => {
         const result = fetchConfig(appConfig, fakeWindow);
         return assertPromiseIsRejected(result, 'Nope');
       });
+
+      it('returns the `groups` array with the initial host config request', async () => {
+        const appConfig = {
+          services: [{ groups: ['group1', 'group2'] }],
+          appType: 'via',
+        };
+        fakeJsonRpc.call.onFirstCall().resolves({ foo: 'baz' }); // host config
+        const result = await fetchConfig(appConfig, fakeWindow);
+        assert.deepEqual(await result.services[0].groups, ['group1', 'group2']);
+      });
+
+      it("creates a merged config where `groups` returns a promise when its initial value is '$rpc:requestGroups'", async () => {
+        const appConfig = {
+          services: [{ groups: '$rpc:requestGroups' }],
+          appType: 'via',
+        };
+        fakeJsonRpc.call.onFirstCall().resolves({ foo: 'baz' }); // host config
+        fakeJsonRpc.call.onSecondCall().resolves(['group1', 'group2']); // requestGroups
+        const result = await fetchConfig(appConfig, fakeWindow);
+        assert.deepEqual(await result.services[0].groups, ['group1', 'group2']);
+        assert.isTrue(
+          fakeJsonRpc.call.getCall(1).calledWithExactly(
+            fakeTopWindow,
+            'https://embedder.com',
+            'requestGroups',
+            [0], // passes service index to requestGroups
+            5000
```

Comment:
Still need to confirm the right timeout value here

Reply:
Any update on this? 5 seconds seems like a reasonable value to start with, but we may need to update it if we learn that it can take longer.